imxrt-hal/src/can/mod.rs

```diff
@@ -0,0 +1,994 @@
+        }
+    }
+
+    fn write_imask_bit(&mut self, mailbox_number: u8, value: bool) {
+        if mailbox_number < 32 {
+            modify_reg!(ral::can, self.reg, IMASK1, |reg| reg
+                | (value as u32) << mailbox_number)
+        } else {
+            modify_reg!(ral::can, self.reg, IMASK2, |reg| reg
+                | (value as u32) << (mailbox_number - 32))
+        }
+    }
+
+    fn read_iflag(&self) -> u64 {
+        (ral::read_reg!(ral::can, self.reg, IFLAG2) as u64) << 32
+            | ral::read_reg!(ral::can, self.reg, IFLAG1) as u64
+    }
+
+    fn read_imask(&self) -> u64 {
+        (ral::read_reg!(ral::can, self.reg, IMASK2) as u64) << 32
+            | ral::read_reg!(ral::can, self.reg, IMASK1) as u64
+    }
+
+    fn write_imask(&mut self, value: u64) {
+        write_reg!(ral::can, self.reg, IMASK1, value as u32);
+        write_reg!(ral::can, self.reg, IMASK2, (value >> 32) as u32);
+    }
+
+    pub fn set_fifo(&mut self, enabled: bool) {
+        self.while_frozen(|this| {
+            modify_reg!(ral::can, this.reg, MCR, RFEN: RFEN_0);
+            this.write_imask(0);
+
+            for i in 0..this.get_max_mailbox() {
+                this.write_mailbox(i, Some(0), Some(0), Some([0x00; 8]));
+                this.write_mailbox_rximr(i, Some(0x00));
+            }
+
+            write_reg!(ral::can, this.reg, RXMGMASK, 0);
+            write_reg!(ral::can, this.reg, RXFGMASK, 0);
+
+            this.write_iflag(this.read_iflag());
+
+            let max_mailbox = this.get_max_mailbox();
+            if enabled {
+                modify_reg!(ral::can, this.reg, MCR, RFEN: RFEN_1);
+                while ral::read_reg!(ral::can, this.reg, MCR, RFEN != RFEN_1) {}
+                for i in this.mailbox_offset()..max_mailbox {
+                    this.write_mailbox(
+                        i,
+                        Some(FlexCanMailboxCSCode::TxInactive.to_code_reg()),
+                        None,
+                        None,
+                    );
+                    this.enable_mailbox_interrupt(i, true);
+                }
+            } else {
+                for i in 0..max_mailbox {
+                    if i < max_mailbox / 2 {
+                        let code = FlexCanMailboxCSCode::RxEmpty.to_code_reg() | 0x00400000 | {
+                            if i < max_mailbox / 4 {
+                                0
+                            } else {
+                                0x00200000
+                            }
+                        };
+                        this.write_mailbox(i, Some(code), None, None);
+                        let eacen = read_reg!(ral::can, this.reg, CTRL2, EACEN == EACEN_1);
+                        let rximr = 0_32 | {
+                            if eacen {
+                                1_u32 << 30
+                            } else {
+                                0
+                            }
+                        };
```

Comment:
`clippy` finds `0_32` curious. Is `0_32` intended to be `0u32` or `32`? If it's just a zero, it looks like we could remove the bitwise OR.

```rust
let rximr: u32 = if eacen { 1 << 30 } else { 0 };
```